test/functional/fixtures/runner/browser-dialogs/before-unload-dialog.test.js

```diff
@@ -0,0 +1,31 @@
+'@fixture before-unload-dialog';
+'@page ./index.html';
+
+
+'@test'['Before unload dialog - should succeed'] = {
+    '1.Click submit button "Enable..."': function () {
+        act.click('#enableBeforeUnload');
+    },
+
+    '2.Click link "This page"': function () {
+        handleBeforeUnload();
+        act.click(':containsExcludeChildren(This page)');
+    }
+};
+
+'@test'['Unexpected before unload dialog - should be failed'] = {
+    '1.Click submit button "Enable..."': function () {
+        act.click('#enableBeforeUnload');
+    },
+
+    '2.Click link "This page"': function () {
+        act.click(':containsExcludeChildren(This page)');
+    }
+};
+
+'@test'['No expected before unload dialog - should be failed'] = {
```

Comment:
Replace with "The expected beforeUnload dialog does not appear. The test should fail."